src/transport/raw/TCP.cpp

```diff
@@ -202,7 +202,7 @@ CHIP_ERROR TCPBase::SendMessage(const Transport::PeerAddress & address, System::
 
     VerifyOrReturnError(address.GetTransportType() == Type::kTcp, CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(mState == TCPState::kInitialized, CHIP_ERROR_INCORRECT_STATE);
-    VerifyOrReturnError(kPacketSizeBytes + msgBuf->DataLength() <= std::numeric_limits<uint16_t>::max(),
+    VerifyOrReturnError(kPacketSizeBytes + msgBuf->DataLength() <= std::numeric_limits<uint32_t>::max(),
```

Comment:
This no longer does the right things with overflow, yes?  (That breakage may predate this PR, and maybe we are not worried about overflow... but on a 32-bit system this test here is always true which is again a huge red flag for me: either the test should not be there because we never hit this case, or the handling of this case is broken on 32-bit systems.)

Reply:
Changed the check to be a tighter bound based on the SystemConfig definition of the max large buffer size.